### ISSUE TITLE
Adding STM32U5 ADC4 support

### ISSUE THREAD

**Issue by stijn577** [embassy-rs/stm32-data#502]
I've been following the guide to try and get the ADC4 working but I'm a bit confused with the Register cleanup section, if I get a field in a register that is a single bit value, how should I convert it so its a true/false item instead of a useless enum, soame thing for the arrays, I can't find these DeleteEnums, MakeRegisterArray, etc. chiptool transforms. 

Am I missing something obvious here?

**Comment by stijn577** [comment]
update: seems i got it working now, the crate seems to build fine as well. However, the yaml doesn't seem to match any other version inside the yaml's provided already (like adc_v4.yaml). 

I believe the architecture is different, but how would it work if say the ADC12 is implemented for the u5 as well, how do the files get merged together, because the ADCs have (slightly) different maps I believe. I also named the block ADC, but ADC4 might be better. So I'm just a bit confused how that would work. (or if they just get kept in seperate files)

**Comment by stijn577** [comment]
update: I tried to use the registers in code
![image](https://github.com/user-attachments/assets/6fc2abed-9206-454a-a43a-6dc9391c2d9c)
![image](https://github.com/user-attachments/assets/cfbe60c8-c887-4628-81e5-e9ed31d8007f)
But it doesn't seem to do anything when viewing the registers with cubeide. I checked if the address values are correct and they seem to be. The start is 0x46021000 as it's supposed to be and the offsets seem to be fine too. So I'm confused as to why it doesn't seem to be doing anything
